packages/routing-utils/src/types.ts

```diff
@@ -21,7 +21,39 @@ export type HasField = Array<
     }
 >;
 
-export type Source = {
+type MiddlewareIndexed =
+  | {
+      /**
+       * A middleware index in the `middleware` key under the build result
+       */
+      middleware: number;
+      /**
+       * A middleware key within the `output` key under the build result
+       */
+      middlewareKey?: never;
+    }
+  | {
+      /**
+       * A middleware index in the `middleware` key under the build result
+       */
+      middleware?: never;
+      /**
+       * A middleware key within the `output` key under the build result
+       */
+      middlewareKey: string;
+    }
+  | {
+      /**
+       * A middleware index in the `middleware` key under the build result
+       */
+      middleware?: never;
+      /**
+       * A middleware key within the `output` key under the build result
+       */
+      middlewareKey?: never;
+    };
+
+export type Source = MiddlewareIndexed & {
```

Comment:
Does this mean it will look like:
```
{ src: '/.*', middleware: 0 }
```
or
```
{ src: '/.*', middlewareKey: '/_middleware' }
```

Reply:
ahh, mm. it should be the same name like the output object. So probably yeah.